core/Azure.Mcp.Core/src/Extensions/OpenTelemetryExtensions.cs

```diff
@@ -28,6 +28,33 @@ public static class OpenTelemetryExtensions
 
     public static void ConfigureOpenTelemetry(this IServiceCollection services)
     {
+        services.AddOptions<AzureMcpServerConfiguration>()
+            .Configure<IOptions<ServiceStartOptions>>((options, serviceStartOptions) =>
+            {
+                // Assembly.GetEntryAssembly is used to retrieve the version of the server application as that is
+                // the assembly that will run the tool calls.
+                var entryAssembly = Assembly.GetEntryAssembly();
+                if (entryAssembly != null)
+                {
+                    options.Version = AssemblyHelper.GetAssemblyVersion(entryAssembly);
+                }
+
+                // Disable telemetry when support logging is enabled to prevent sensitive data from being sent
+                // to telemetry endpoints. Support logging captures debug-level information that may contain
+                // sensitive data, so we disable all telemetry as a safety measure.
+                if (!string.IsNullOrWhiteSpace(serviceStartOptions.Value.SupportLoggingFolder))
+                {
+                    options.IsTelemetryEnabled = false;
+                    return;
+                }
+
+                // This environment variable can be used to disable telemetry collection entirely. This takes precedence
+                // over any other settings.
+                var collectTelemetry = Environment.GetEnvironmentVariable("AZURE_MCP_COLLECT_TELEMETRY");
+
+                options.IsTelemetryEnabled = string.IsNullOrWhiteSpace(collectTelemetry) || (bool.TryParse(collectTelemetry, out var shouldCollect) && shouldCollect);
+            });
```

Comment:
This code was moved to ServiceCollectionExtensions.cs.  Please rebase and update the configuration there.